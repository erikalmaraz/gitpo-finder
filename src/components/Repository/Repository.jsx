import React from "react";
// import * as S from "./Syles"
const Repository = () => {
  return (
    <>
     <div>
       Repositorio
     </div>
    </>
  );
};

export default Repository;
