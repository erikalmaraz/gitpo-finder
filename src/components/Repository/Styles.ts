import styled from "styled-components";

export const RepoContainer = styled.div`
  min-height: 100px;
  padding: 10px;
  position: relative;
  margin: 12px 0px;
  border: 1px solid #ccc;
  border-radius: 8px;
`;

export const Name = styled.div`
  font-weight: bold;
  font-size: 1.1rem;
`;

export const Description = styled.div`
  margin: 10px 0;
`;

export const Language = styled.div`
  position: absolute;
  bottom: 10px;
  font-size: 0.75rem;
  font-weight: bold;
`;
