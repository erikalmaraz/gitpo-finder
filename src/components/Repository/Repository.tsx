import React from "react";
import RespositoryType from "../../interfaces/repository.interface";
import * as S from "./Styles";
const Repository = ({ name, description, language }: RespositoryType) => {
  return (
    <React.Fragment>
      <S.RepoContainer>
        <S.Name>{name}</S.Name>
        <S.Description>{description}</S.Description>
        <S.Language>{language}</S.Language>
      </S.RepoContainer>
    </React.Fragment>
  );
};

export default Repository;
