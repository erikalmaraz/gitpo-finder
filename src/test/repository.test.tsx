import React from "react";
import { render, screen } from "@testing-library/react";
import Repository from "../components/Repository/Repository";
import renderer from "react-test-renderer";

it("Should render Repository correctly", () => {
  const props = {
    name: "Repository name",
    description: "Repository desc",
    language: "JavaScript",
  };
  const tree = renderer.create(<Repository {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
