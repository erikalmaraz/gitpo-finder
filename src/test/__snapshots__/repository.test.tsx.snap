// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Repository correctly 1`] = `
<div
  className="sc-bdfBQB kkSnTO"
>
  <div
    className="sc-gsTEea hXCRma"
  >
    Repository name
  </div>
  <div
    className="sc-dlfnuX lehMsm"
  >
    Repository desc
  </div>
  <div
    className="sc-hKgJUU hGsbJj"
  >
    JavaScript
  </div>
</div>
`;
